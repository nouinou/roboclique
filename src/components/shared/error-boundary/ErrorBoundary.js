import { Component } from 'react';

const ERROR_MESSAGE =
	'Something went wrong! Please try again later or contact the developer';

class ErrorBoundary extends Component {
	constructor(props) {
		super(props);
		this.state = {
			hasError: false,
		};
	}

	componentDidCatch() {
		this.setState({ hasError: true });
	}

	render() {
		if (this.state.hasError) {
			return (
				<div className="flex flex-col min-h-content justify-center items-center">
					<div className="robo-text text-2xl sm:text-3xl text-red-500 max-w-lg">
						{ERROR_MESSAGE}
					</div>
				</div>
			);
		} else {
			return this.props.children;
		}
	}
}

export default ErrorBoundary;
