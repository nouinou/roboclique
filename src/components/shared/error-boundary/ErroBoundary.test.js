import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary Component', () => {
	it('should render ErrorBoundary', () => {
		expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
	});
});
