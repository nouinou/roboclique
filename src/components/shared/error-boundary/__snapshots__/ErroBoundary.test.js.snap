// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component should render ErrorBoundary 1`] = `""`;
